### PR TITLE
Add check for urls present in requires_dist

### PR DESCRIPTION
**Pull request by oliverholworthy**
- [We failed to publish the pypi pacakge in the last release due to invalid values in our package requires_dist](https://github.com/NVIDIA-Merlin/NVTabular/actions/runs/3528066078/jobs/5918246198)
- This PR adds a check to our build process to return an error if we have urls in our package `requires_dist`

### CODE REVIEW

**Comment by oliverholworthy** [comment]
I'm expecting the `CPU CI / build` to fail on this new check. And then hopefully pass after #1727 is merged in

**Review by mikemckiernan — APPROVED** [review]
You're welcome to request a review from someone more skilled in CI, but I certainly approve the idea of adding a test after we've found a problem.  Thanks.

**Comment by oliverholworthy** [comment]
Updated with an error message if this fails. And moved the check to a dedicated script in `./ci/check_dist.sh`

Example Failure: https://github.com/NVIDIA-Merlin/NVTabular/actions/runs/3697897020/jobs/6263443772#step:7:86

image

**Pull request merged**
